File: Santiago/MiningMinoc.pas
(********************************************)
(*************** HEADER    ******************)
(********************************************)
Program Mining;
(*************** Units     ******************)
Uses
  Sysutils,
  Constants,
  BankUtils;
(*************** Structures******************)
(*************** Constants ******************)
const
    mine_X = 2580;
    mine_Y = 450;
    bank_X = 2505;
    bank_Y = 542;
    storageContainerType = $09AA;
(*************** Variables ******************)
var
    pickaxe: Cardinal;
(*************** Methods   ******************)
procedure Setup();
begin
    AddToDebugJournal('Setting Up ...');
    Disarm();
    Wait(1000);
    pickaxe := FindType(pickaxeType, Backpack);
    AddToDebugJournal('Done.');
end;

function CheckWeight(): Boolean;
begin
    Result := Weight < (MaxWeight - 30);
end;

procedure MineTile(x: Integer; y: Integer);
var
    startTime: TDateTime;
    option: Cardinal;
begin
    AddToDebugJournal('  Mining (' + IntToStr(x) + ', ' + IntToStr(y) + ') ...');
    repeat
        if TargetPresent then CancelTarget;
        Disarm();  
        UseObject(FindType(pickaxeType,Backpack));
        WaitTargetXYZ(x, y, GetZ(Self))
        startTime := Now();
        WaitJournalLineSystem(startTime, 'line|elsewhere|nothing|close|decide|away|rocks|put', 10000);
        option := FoundedParamId;
        AddToDebugJournal('    ' + Journal(LineIndex()));
    until option <= 5;
end;

procedure MineAround();
var
    x, y, x_idx, y_idx : Integer;
begin                                                          
    x := GetX(Self);
    y := GetY(Self);
    x_idx := -2;
    y_idx := -2;
    AddToDebugJournal('  Starting to Mine around (' + IntToStr(x) + ', ' + IntToStr(y) + ') ...');
    while (x_idx < 3) do
    begin
        while (y_idx < 3) and (CheckWeight()) do
        begin
            MineTile(x + x_idx, y + y_idx);
            y_idx := y_idx + 1;
        end;
        x_idx := x_idx + 1;
        y_idx := -2;
    end;
    AddToDebugJournal('  Done Mining around (' + IntToStr(x) + ', ' + IntToStr(y) + ').');
end;

procedure Move();
var
    new_x, new_y: Integer;
begin
    AddToDebugJournal('  Relocating randomly ...');
    repeat
        new_x := GetX(Self) + (2 * RandomRange(-2, 2));
        new_y := GetY(Self) + (2 * RandomRange(-2, 2));
    until (new_x <> GetX(Self)) or (new_y <> GetY(Self))
    MoveXY(new_x, new_y, True, 1, True);
    AddToDebugJournal('  Done relocating to (' + IntToStr(GetX(Self)) + ', ' + IntToStr(GetY(Self)) + ').');
end;

procedure Mine();
begin
    AddToDebugJournal('Starting to Mine ...');
    while CheckWeight() do
    begin
        Move();
        MineAround();
    end;
    AddToDebugJournal('Done Mining ...');
end;

procedure GoToMine();
begin
    AddToDebugJournal('Going to the mine ...');
    MoveXY(mine_X, mine_Y, True, 3, True);
    AddToDebugJournal('Arrived at the mine (' + IntToStr(GetX(Self)) + ', ' + IntToStr(GetY(Self)) + ')');
end;

procedure GoToBank();
begin
    AddToDebugJournal('Going to the bank ...');
    MoveXY(bank_X, bank_Y, True, 1, True);
    AddToDebugJournal('Arrived at the bank (' + IntToStr(GetX(Self)) + ', ' + IntToStr(GetY(Self)) + ')');
end;

procedure StoreOres();
begin
    StoreItemsAtBankContainer(orePileTypes, storageContainerType);
    StoreItemsAtBankContainer(gemStoneTypes, storageContainerType);
end;

begin
    Setup();
    repeat
        GoToBank();
        StoreOres();
        GoToMine();
        Mine();
    until Dead;
end.

File: TFG/Stealth/MiningMinoc.py
from py_stealth import *
from Constants import *
from datetime import datetime
import random

mine_X = 2580
mine_Y = 450
bank_X = 2505
bank_Y = 542

def check_weight():
    return Weight() < (MaxWeight() - 30)


def mine_tile(x, y):
    option = 0

    while option <= 1:
        if TargetPresent():
            CancelTarget()
        Disarm()
        UseObject(FindType(PICKAXE_TYPE, Backpack()))
        WaitTargetXYZ(x, y, GetZ(Self()))
        WaitJournalLineSystem(datetime.now(), 'put|rocks|line|elsewhere|nothing|close|decide|away', 10000)
        option = FoundParamID()

def mine_around():
    for x_idx in range(-2, 3):
        for y_idx in range(-2, 3):
            if check_weight():
                mine_tile(GetX(Self()) + x_idx, GetY(Self()) + y_idx)

def move():
    new_x = 0
    new_y = 0

    while new_x == 0 and new_y == 0:
        new_x = GetX(Self()) + (2 * random.randint(-2, 2))
        new_y = GetY(Self()) + (2 * random.randint(-2, 2))

    MoveXY(new_x, new_y, True, 1, True)

def mine():
    while check_weight():
        move()
        mine_around()

def go_to_mine():
    MoveXY(mine_X, mine_Y, True, 3, True)

def go_to_bank():
    MoveXY(bank_X, bank_Y, True, 1, True)

def store_ores():


while not Dead():
    go_to_bank()
    store_ores()
    go_to_mine()
    mine()